inspirehep/modules/workflows/tasks/actions.py

```diff
@@ -1149,3 +1149,27 @@ def load_record_from_hep(obj, wf):
     obj.extra_data['head_revision_id'] = record_data['revision_id']
 
     return obj
+
+
+def remove_inspire_categories_derived_from_core_arxiv_categories(obj, eng):
+    if not obj.data.get("arxiv_eprints"):
+        return
+    inspire_categories_without_arxiv_sourced = [
+        category
+        for category in obj.data.get("inspire_categories")
+        if category.get("source") != "arxiv"
+    ]
+    non_core_arxiv_categories = [
+        arxiv_category
+        for arxiv_category in obj.data["arxiv_eprints"][0]["categories"]
```

Comment:
are we sure that we get always `arxiv_eprints[0].categories`?

Reply:
yes, you are right, `categories` are not required in `arxiv_eprints` 👍